src/router_core/connections.c

```diff
@@ -1312,78 +1312,81 @@ void qdr_check_addr_CT(qdr_core_t *core, qdr_address_t *addr)
 
 static void qdr_connection_opened_CT(qdr_core_t *core, qdr_action_t *action, bool discard)
 {
-
     qdr_connection_t *conn = safe_deref_qdr_connection_t(action->args.connection.conn);
-    if (!discard && conn) {
 
-        do {
-            DEQ_ITEM_INIT(conn);
-            DEQ_INSERT_TAIL(core->open_connections, conn);
+    if (!conn)
```

Comment:
Oh wait - if the conn has been freed whilst the action was queued conn will be null but connection_label and container_id will still need to be freed, right?

Reply:
Agreed, new commit pushed to take care of that case. thanks much